lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -124,14 +127,30 @@ def add_shopper_interaction(post, payment, options={})
       def add_address(post, options)
         return unless post[:card] && post[:card].kind_of?(Hash)
         if (address = options[:billing_address] || options[:address]) && address[:country]
```

Comment:
_Nothing to do with code logic, just code cleanup:_

I'd prefer if we didn't do any assignments in conditionals; same goes for lower down. What would you think of having:

```
return unless post[:card] && post[:card].kind_of?(Hash)

address = options[:billing_address] || options[:address]
shipping_address = options[:shipping_address] || address
```